sdks/python/apache_beam/io/external/xlang_jdbcio_it_test.py

```diff
@@ -0,0 +1,165 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# pytype: skip-file
+
+from __future__ import absolute_import
+
+import logging
+import typing
+import unittest
+
+from past.builtins import unicode
+
+import apache_beam as beam
+from apache_beam import coders
+from apache_beam.io.external.jdbc import WriteToJdbc
+from apache_beam.options.pipeline_options import PipelineOptions
+from apache_beam.testing.test_pipeline import TestPipeline
+
+# pylint: disable=wrong-import-order, wrong-import-position
+try:
+  import psycopg2
+except ImportError:
+  unittest.skip("no psycopg2 installed")
+  # pylint: enable=wrong-import-order, wrong-import-position
+
+JdbcTestRow = typing.NamedTuple(
+    "JdbcTestRow",
+    [
+        ("f_id", int),
+        ("f_real", float),
+        ("f_string", unicode),
+    ],
+)
+
+coders.registry.register_coder(JdbcTestRow, coders.RowCoder)
+
+ROW_COUNT = 10
+
+
+class JdbcExternalTransformTest(unittest.TestCase):
+  """Tests that exercise the cross-language JdbcIO Transform (implemented in java).
+
+  To run with the local expansion service and flink job server you need to build it,
+  e.g. via command:
+  ./gradlew :sdks:java:io:expansion-service:shadowJar
+  ./gradlew :runners:flink:1.10:job-server:shadowJar
+  and have flink1.10 master cluster running
+
+  If on development branch, it may be necessary to build java sdk docker image and tag
+  it with the latest released version.
+  ./gradlew :sdks:java:container:docker
+  docker tag apache/beam_java_sdk:x.xx.x.dev apache/beam_java_sdk:y.yy.y.dev
+
+  python setup.py nosetests \
+  --tests=apache_beam.io.external.jdbc_test:JdbcExternalTransformTest.test_read_and_write \
+  --test-pipeline-options="
+    --driver_class_name=org.postgresql.Driver
+    --jdbc_url=jdbc:postgresql://localhost:5432/postgres
```

Comment:
This would require the user to setup an instance of postgres with these parameters right? I think you should have this test create the postgres instance it needs, I bet there are python libraries that can help with that, e.g. https://pypi.org/project/pytest-postgresql/

Reply:
I didn't think of it as I thought that there is some postgres instance running where this test could be executed. I will take a look how to easily instantiate the postgres image and try to make this test click and run.

Reply:
I used https://github.com/testcontainers/testcontainers-python and now postgres instance is instantiated automatically. Now the test can be run without any params

Reply:
Thanks!